packages/plugins/jwt/src/config.ts

```diff
@@ -0,0 +1,141 @@
+import { VerifyOptions } from 'jsonwebtoken';
+import { Options as JwksClientOptions } from 'jwks-rsa';
+
+type DiscriminatedUnion<K extends PropertyKey, T extends object> = {
+  [P in keyof T]: { [Q in K]: P } & T[P] extends infer U ? { [Q in keyof U]: U[Q] } : never;
+}[keyof T];
+
+type AtleastOneItem<T> = [T, ...T[]];
+
+type ExtractTokenFunction = (params: {
+  request: Request;
+  serverContext: object | undefined;
+  url: URL;
+}) => Promise<string | undefined> | string | undefined;
+
+export type TokenLookupLocation = DiscriminatedUnion<
+  'type',
+  {
+    header: {
+      name: string;
+      prefix?: string;
+    };
+    cookie: {
+      name: string;
+    };
+    fn: {
+      getToken: ExtractTokenFunction;
+    };
+  }
+>;
+
+export type JwksSource = DiscriminatedUnion<
+  'type',
+  {
+    remote: { jwksClientOptions: JwksClientOptions };
+    inline: { signingKey: string };
+  }
+>;
+
+export type JwtPluginOptions = {
+  /**
+   * List of configurations for the signin-key providers. You can configure multiple signin-key providers to allow for key rotation, fallbacks, etc.
+   *
+   * In addition, you can use the `remote` variant and configure [`jwks-rsa`'s JWKS client](https://github.com/auth0/node-jwks-rsa/tree/master).
+   *
+   * The plugin will try to fetch the keys from the providers in the order they are defined in this array.
+   *
+   * If the first provider fails to fetch the keys, the plugin will try the next provider in the list.
+   *
+   */
+  singingKeyProviders: AtleastOneItem<JwksSource>;
+  /**
+   * List of locations to look for the token in the incoming request.
+   *
+   * By defualt, the plugin will look for the token in the `Authorization` header with the `Bearer` prefix.
+   *
+   * The plugin will try to extract the token from the locations in the order they are defined in this array.
+   *
+   * If the token is found in one of the locations, the plugin will stop looking for the token in the other locations.
+   *
+   * If the token is not found in any of the locations, the plugin will mark the authentication as failed.
+   *
+   * Can be used with `rejectUnauthenticatedRequests: { missingToken: true }` to reject requests without a token.
+   *
+   */
+  tokenLookupLocations?: AtleastOneItem<TokenLookupLocation>;
```

Comment:
You don't think we could have a sensible default here ? Like seeking for the `authorization` Bearer token ? And the `authorization` cookie ? Those are the most common places where they can be found.

Reply:
There's a default value configured for this one (`Authorization` header with `Bearer` token)